### DISCUSSION TITLE
Is it possible to print month's name instead of number?

### DISCUSSION THREAD

**Discussion by DanteNahuel** [Makuna/Rtc#127] [Q&A] [answered]
Does the library allow this or is this not implemented?

Thank you

**Comment by Makuna** [comment, marked as answer]
It is not implemented and left up to the sketch writer.

Storing the strings representing months has two major issues the sketch writer has to be concerned with:

1) The string names of the months take up code space if implemented with PROGMEM feature.  Otherwise depending on compiler they take up valuable RAM.
2) Exact names to use is sketch dependent; not only full length name like "January" but also abbreviated names like "Jan" and on top of that, localization for nationality/language it is being written for.

**Reply by DanteNahuel** [reply]
Thank you for taking the time to reply this.
I understand point 2.
However, when discussing point 1, if the MCU where this is used is an ESP32, this shouldn't be an issue, right?

**Reply by Makuna** [reply]
Its less of a problem, but almost never not a problem ;-)

If you need to display the names, you have to store them somehow.

Almost always use progmem, RAM is more precious.
```
const char c_MonthNameDecemberary[] PROGMEM = "Decemberary";
```
Do a search for PROGMEM Arduino and it will give you all the details on how to define and access them.  They are almost always as fast as RAM but do require specific functions.

**Reply by DanteNahuel** [reply]
Thank you again for the complete reply!
